hs_core/irods.py

```diff
@@ -0,0 +1,246 @@
+import os
+
+from datetime import datetime, timedelta
+
+from django.db import models
+from django.core.exceptions import PermissionDenied, ValidationError
+from mezzanine.conf import settings
+
+from hs_core.signals import pre_check_bag_flag
+
+
+class ResourceIRODSMixin(models.Model):
+    """ This contains iRODS methods to be included as options for resources """
+    class Meta:
+        abstract = True
+
+    def __home_path(self):
+        """ Return the home path for local iRODS resources """
+        return settings.IRODS_CWD
+
+    def update_bag(self):
+        """
+        Update a bag if necessary.
+
+        This uses the Django signal pre_check_bag_flag to prepare collections, i
+        and then checks the AVUs 'metadata_dirty' and 'bag_modified' to determine
+        whether to regenerate the metadata files and/or bag.
+
+        This is a synchronous update. The call waits until the update is finished.
+        """
+        from hs_core.tasks import create_bag_by_irods
+        from hs_core.hydroshare.resource import check_resource_type
+        from hs_core.hydroshare.hs_bagit import create_bag_files
```

Comment:
Putting them at the beginning of the file creates an import loop. That's why they're here!